index.js

```diff
@@ -22,394 +23,483 @@ function str2hex(str) {
 	return result;
 }
 
-function instance(system, id, config) {
-	var self = this;
+class instance extends instance_skel {
+	constructor(system, id, config) {
+		super(system, id, config);
+
+		this.awaiting_reply = false;
+		this.command_queue = [];
+		this.files = [];
+		this.last_bit_check = null;
+		this.current_transport_status = null;
+
+		// Number of ms to grab transport updates
+		this.defineConst('TRANSPORT_UPDATES', 2000);
+		// If we miss x status updates, let's attempt to reconnect
+		this.defineConst('TCP_TIMEOUT', this.TRANSPORT_UPDATES * 5);
+
+		this.transport_bits = {
+			PLAY: {
+				bit: 0,
+				name: 'Playing',
+			},
+			RECORD: {
+				bit: 1,
+				name: 'Recording',
+			},
+			FF: {
+				bit: 2,
+				name: 'FF',
+			},
+			RW: {
+				bit: 3,
+				name: 'RW',
+			},
+			STOP: {
+				bit: 5, // Warning: this seems to always be 0 although documentation says it will be set
+				name: 'Stopped'
+			}
+		};
 
-	self.awaiting_reply = false;
-	self.command_queue = [];
-	self.files = [];
+		this.actions(); // export actions
 
-	// super-constructor
-	instance_skel.apply(this, arguments);
+		return this;
```

Comment:
`return this;` shouldn't be necessary in the constructor, though I don't believe there's a penalty for it either.